test/acceptance/tests/connect/connect_inject_test.go

```diff
@@ -34,37 +34,55 @@ func TestConnectInjectDefault(t *testing.T) {
 // Test that Connect works in a secure installation,
 // with ACLs and TLS enabled.
 func TestConnectInjectSecure(t *testing.T) {
-	cfg := suite.Config()
-	ctx := suite.Environment().DefaultContext(t)
-
-	helmValues := map[string]string{
-		"connectInject.enabled":        "true",
-		"global.tls.enabled":           "true",
-		"global.acls.manageSystemACLs": "true",
+	cases := []struct {
+		name              string
+		enableAutoEncrypt string
+	}{
+		{
+			"without auto-encrypt",
+			"false",
+		},
+		{
+			"with auto-encrypt",
+			"true",
+		},
+	}
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			cfg := suite.Config()
+			ctx := suite.Environment().DefaultContext(t)
+
+			helmValues := map[string]string{
+				"connectInject.enabled":        "true",
+				"global.tls.enabled":           "true",
+				"global.tls.enableAutoEncrypt": c.enableAutoEncrypt,
+				"global.acls.manageSystemACLs": "true",
+			}
+
+			releaseName := helpers.RandomName()
+			consulCluster := framework.NewHelmCluster(t, helmValues, ctx, cfg, releaseName)
+
+			consulCluster.Create(t)
+
+			t.Log("creating static-server and static-client deployments")
+			helpers.Deploy(t, ctx.KubectlOptions(), cfg.NoCleanupOnFailure, "fixtures/static-server.yaml")
+			helpers.Deploy(t, ctx.KubectlOptions(), cfg.NoCleanupOnFailure, "fixtures/static-client.yaml")
+
+			t.Log("checking that the connection is not successful because there's no intention")
+			helpers.CheckStaticServerConnection(t, ctx.KubectlOptions(), "static-client", false, "http://localhost:1234")
```

Comment:
same thing about constants for "static-client" and "static-server". Also, does it make sense to add a comment about why we are testing the connection against "http://localhost:1234"?